src/jit/codegenlegacy.cpp

```diff
@@ -20853,22 +20867,21 @@ GenTreePtr Compiler::fgLegacyPerStatementLocalVarLiveness(GenTreePtr startNode,
                 // This ensures that the block->bbVarUse will contain
                 // the FrameRoot local var if is it a tracked variable.
 
-                if (tree->gtCall.IsUnmanaged() || (tree->gtCall.IsTailCall() && info.compCallUnmanaged))
-                {
-                    /* Get the TCB local and mark it as used */
-
-                    noway_assert(info.compLvFrameListRoot < lvaCount);
+                if (info.compLvFrameListRoot < lvaCount)
```

Comment:
@jkotas I've tried this variant before but got some fails on testing. Ok, I'll try it again.